### PR TITLE
Updated Dev Guide to describe current CI workflows

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR updates Dev Guide documentation to describe or CI workflows for Azure and GitHub Actions which were recently modified.

The generated RTD documentation for this PR branch is here: https://raja.readthedocs.io/en/task-rhornung67-devguide-update/sphinx/dev_guide/index.html

### CODE REVIEW

**Comment by rhornung67** [comment]
@adrienbernede do have any thoughts on what's going on with the RAJA poodle jobs? They have been dying in the resource allocation part since last week. I posted a message in LLNL GitLab channel on Mattermost, but no response yet.

**Comment by adrienbernede** [comment]
> @adrienbernede do have any thoughts on what's going on with the RAJA poodle jobs? They have been dying in the resource allocation part since last week. I posted a message in LLNL GitLab channel on Mattermost, but no response yet.

I also asked yesterday. Although it looked like a failing allocation, the job log suggested a runner issue. I checked that poodle was up before reaching out for help. It should be OK now, but one less runner increases the competition.

**Comment by rhornung67** [comment]
Folks, this is ready to merge. Please review.

**Comment by rhornung67** [comment]
@adrienbernede there is a response on mattermost to my post. it looks like they don't understand the issue with the poodle18 runner.

File: docs/sphinx/dev_guide/ci.rst [hunk 1]
```diff
@@ -17,14 +17,17 @@ Continuous Integration (CI) Testing
...
-    `RSE Ops Project <https://github.com/rse-ops/docker-images>`_.
-    While we do some GPU builds on Azure, RAJA tests are only run for CPU-only 
-    builds. The current set of builds run on Azure can be seen by looking at
-    the ``RAJA/azure-pipelines.yml`` and ``RAJA/Dockerfile`` files.
+The RAJA project uses threes CI tools to run tests:
+
+  * **Azure Pipelines** and **GitHub Actions** runs builds and tests for Linux,
+    Windows, and MacOS environments using compilers in container images 
+    maintained in the `RADIUSS Docker Project <https://github.com/LLNL/radiuss-docker>`_.
+    While we do some GPU builds on Azure and GitHub Actions, for CUDA, HIP, and
```
Comment (MrBurmark) [inline docs/sphinx/dev_guide/ci.rst:25]:
```suggestion
    While we do some GPU builds on Azure and GitHub Actions for CUDA, HIP, and
```

File: docs/sphinx/dev_guide/ci_tasks.rst [hunk 2]
```diff
@@ -291,78 +317,100 @@ If we want to add a new Azure pipeline to build with ``compilerX``, then in ``az
...
-      cd .. && rm -rf build
+      make clean
 
-Each of our docker builds is built up on a base image maintained by RSE-Ops, a table of available base containers can be found `here <https://rse-ops.github.io/docker-images/>`_. We are also able to add target names to each build with ``AS ...``. This target name correlates to the ``docker_target: ...`` defined in ``azure-pipelines.yml``.
+Each of our docker builds is built up on a base image maintained in the
+`RADIUSS Docker Project <https://github.com/LLNL/radiuss-docker>`_.
 
-The base containers are shared across multiple projects and are regularly rebuilt. If bugs are fixed in the base containers the changes will be automatically propagated to all projects using them in their Docker builds.
+The base container images are shared by multiple projects and are rebuilt
+regularly rebuilt. If bugs are fixed in the base images, the changes will be
```
Comment (MrBurmark) [inline docs/sphinx/dev_guide/ci_tasks.rst:349]:
```suggestion
The base container images are shared by multiple projects and are rebuilt
regularly. If bugs are fixed in the base images, the changes will be
```

File: docs/sphinx/dev_guide/ci.rst (hunk 1 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst]: Fixed.

File: docs/sphinx/dev_guide/ci_tasks.rst (hunk 2 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci_tasks.rst]: Fixed.

**Review by MrBurmark — APPROVED** [review]

File: docs/sphinx/dev_guide/ci.rst [hunk 3]
```diff
@@ -17,14 +17,17 @@ Continuous Integration (CI) Testing
...
 
-The RAJA project uses two CI tools to run tests:
-
-  * **Azure Pipelines** runs builds and tests for Linux, Windows, and MacOS 
-    environments using compilers in container images maintained in the
-    `RSE Ops Project <https://github.com/rse-ops/docker-images>`_.
-    While we do some GPU builds on Azure, RAJA tests are only run for CPU-only 
-    builds. The current set of builds run on Azure can be seen by looking at
-    the ``RAJA/azure-pipelines.yml`` and ``RAJA/Dockerfile`` files.
+The RAJA project uses threes CI tools to run tests:
```
Comment (long58) [inline docs/sphinx/dev_guide/ci.rst:20]: Should this be "uses three CI tools" instead of "threes"?

Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:20]: Good catch. Fixed.

**Review by long58 — APPROVED** [review]

**Pull request merged**